### PR TITLE
feat: add restricted paths to ingress

### PR DESCRIPTION
**Pull request by deontpearson**
## Description

- Provide the ability for the client to control the ingress annotations when it is created.
- Legacy ingress types do not use the `ingress.className` value. It must be controlled through the annotations field

## Types of changes

_Put an `x` in the boxes that apply._

- [ ] Bugfix (`fix`): A non-breaking change which fixes an issue or bug
- [x] New Feature (`feat`): A non-breaking change which adds a new feature or new functionality
- [ ] Tests (`test`): Adding missing tests or correcting existing tests
- [ ] Refactoring (`refactor`): A code change that neither fixes a bug nor adds a feature, e.g Deleting unused or redundant code
- [ ] Documentation (`docs`): Documentation only changes
- [ ] Other (`style`, `build`, `ci`, `performance`): Formatting, linting or styling. Changes to the CI pipeline. Performance improvements. Changes that affect the build system.

### CODE REVIEW

File: examples/aruba-uxi-example/values-staging.yaml [hunk 1]
```diff
@@ -79,8 +79,10 @@ aruba-uxi:
         example-service-staging: example-service-staging
       ingress:
         className: internal-gateway
+        internalPaths:
```
Comment (brandon-piner-aruba) [inline examples/aruba-uxi-example/values-staging.yaml:82]: maybe 
```suggestion
        externalPaths:
```

**Comment by brandon-piner-aruba** [comment]
May be better the move the snippet to the values file so that it is always up to the client to implement

File: .yamllint.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 extends: default
 
 rules:
-  line-length:
-    max: 100
```
Comment (1riatsila1) [inline .yamllint.yml:5]: Why are we changing this?

Comment (deontpearson) [reply .yamllint.yml:5]: There was a discussion a few weeks back where the conclusion was to do away with line length in yaml files. Cookiecutter has this change so I just added it here. Do you disagree?

File: charts/aruba-uxi/templates/application-ingress.yaml [hunk 3]
```diff
@@ -12,9 +12,9 @@ metadata:
   name: {{ $name }}
   labels:
     {{- include "aruba-uxi.labels" (dict "context" $ "name" $name "additionalLabels" $additionalLabels) | nindent 4 }}
-  {{- if $ingress.className }}
+  {{- with $ingress.annotations }}
   annotations:
-    kubernetes.io/ingress.class: {{ $ingress.className }}
+    {{- toYaml . | nindent 4 }}
```
Comment (brandon-piner-aruba) [inline charts/aruba-uxi/templates/application-ingress.yaml:17]:
```suggestion
  annotations:
    kubernetes.io/ingress.class: {{ $ingress.className }}
    {{- with $ingress.annotations }}
    {{- toYaml . | nindent 4 }}
```

**Review by brandon-piner-aruba — APPROVED** [review]

File: .yamllint.yml (hunk 2 above)
Comment (1riatsila1) [reply .yamllint.yml:5]: Hmmmm, yeah I think I would prefer line length - but like... it is not a big deal at all so I am super happy to do away with it if that is what a previous discussion concluded :-)

File: charts/aruba-uxi/templates/application-ingress.yaml (hunk 3 above)
Comment (deontpearson) [reply charts/aruba-uxi/templates/application-ingress.yaml:17]: done

**Pull request merged**